video_worker/tests/test_generate_encode.py

```diff
@@ -0,0 +1,117 @@
+        """
+        Tests settings_setup works correctly.
+        """
+
+        def change_worker_setup(self):
+            """
+            Sets workerSetup.instance_yaml to TEST_INSTANCE_YAML when WorkerSetup is initiallized.
+            """
+            self.instance_yaml = TEST_INSTANCE_YAML
+            self.setup = False
+
+        # Setup WS.settings_dict
+        WS = create_worker_setup()
+        WS.run()
+
+        with patch('video_worker.config.WorkerSetup.__init__', new=change_worker_setup) as ws_mock:
+            result_settings_dict = self.command_generate.settings_setup()
+            self.assertEqual(result_settings_dict, WS.settings_dict)
+
+    @data(
+        (
+            {
+                'video_object': None,
+                'error_message': 'Command Gen Fail: No Video Object'
+            }
+        ),
+        (
+            {
+                'encode_object': None,
+                'error_message': 'Command Gen Fail: No Encode Object'
+            }
+        ),
+        (
+            {
+                'job_id': 'dummy-job-id',
+                'ENFORCE_TARGET_ASPECT': True
+            }
+        ),
+    )
+    @patch('video_worker.generate_encode.logger')
+    @patch.object(CommandGenerate, '_destination')
+    @patch.object(CommandGenerate, '_passes')
+    @patch.object(CommandGenerate, '_scalar')
+    @patch.object(CommandGenerate, '_codec')
+    @patch.object(CommandGenerate, '_call')
+    def test_generate(self, mock_data, mock_call, mock_codec, mock_scalar, mock_passes, mock_destination, mock_logger):
+        """
+        Tests `generate` method works correctly.
+        """
+        ffcommand = ['dummy-ffcommand-arg-1', 'dummy-ffcommand-arg-2']
+        job_id = mock_data.get('job_id', None)
+        command_generate = CommandGenerate(
+            VideoObject=mock_data.get('video_object', self.video),
+            EncodeObject=mock_data.get('encode_object', self.encode),
+            jobid=job_id
+        )
+
+        self.assertEqual(command_generate.ffcommand, [])
+        self.assertIsNone(command_generate.workdir)
+
+        command_generate.ffcommand = ffcommand
+        result_command = command_generate.generate()
+
+        if mock_data.get('error_message', ''):
+            mock_logger.error.assert_called_with(mock_data.get('error_message'))
+        else:
+            self.assertIsNotNone(command_generate.workdir)
+            expected_workdir = os.path.join(ENCODE_WORK_DIR, job_id) if job_id else ENCODE_WORK_DIR
+            self.assertEqual(command_generate.workdir, expected_workdir)
+            self.assertEqual(result_command, ' '.join(ffcommand))
+
+            if mock_data.get('ENFORCE_TARGET_ASPECT', False):
+                self.assertTrue(mock_scalar.called)
+
+            self.assertTrue(mock_call.called)
+            self.assertTrue(mock_codec.called)
+            self.assertTrue(mock_passes.called)
+            self.assertTrue(mock_destination.called)
```

Comment:
Everything is mocked here, can not we test the `generate` method without mocking. As per my understanding `generate` method is not doing any third party calls.